src/neo/SmartContract/ApplicationEngine.Contract.cs

```diff
@@ -28,22 +28,29 @@ partial class ApplicationEngine
         /// </summary>
         public static readonly InteropDescriptor System_Contract_CreateStandardAccount = Register("System.Contract.CreateStandardAccount", nameof(CreateStandardAccount), 0_00010000, CallFlags.None, true);
 
-        protected internal void CreateContract(byte[] script, byte[] manifest)
+        protected internal void CreateContract(byte[] nefFile, byte[] manifest)
         {
-            if (script.Length == 0 || script.Length > MaxContractLength)
-                throw new ArgumentException($"Invalid Script Length: {script.Length}");
+            if (nefFile.Length == 0)
+                throw new ArgumentException($"Invalid NefFile");
             if (manifest.Length == 0 || manifest.Length > ContractManifest.MaxLength)
                 throw new ArgumentException($"Invalid Manifest Length: {manifest.Length}");
 
-            AddGas(StoragePrice * (script.Length + manifest.Length));
+            NefFile nef = nefFile.AsSerializable<NefFile>();
+            if (nef.Script.Length > MaxContractLength)
+                throw new ArgumentException($"Invalid Script Length: {nef.Script.Length}");
+            if (nef.Abi.Length == 0 || nef.Abi.Length > ContractAbi.MaxLength)
+                throw new ArgumentException($"Invalid Abi Length: {nef.Abi.Length}");
 
-            UInt160 hash = script.ToScriptHash();
+            AddGas(StoragePrice * (nef.Script.Length + nef.Abi.Length + manifest.Length));
+
+            UInt160 hash = nef.Script.ToScriptHash();
```

Comment:
It should be the hash of NefFile.

Reply:
Done, native contracts script hash should be the same as before, right?
Should we check that the nef has been fully read? it could be altered in hash by adding `0x00` to the end of the nef file